### ISSUE TITLE
Expose number of samples for emitting denoised copy ratios in gCNV or optimize sampling method.

### ISSUE THREAD

**Issue by samuelklee** [broadinstitute/gatk#5754]

**Comment by samuelklee** [comment]
Sampling method optimized in #5781.

**Comment by samuelklee** [comment]
Default changed from 250 -> 20 in #5699 and exposed in #7450. Unfortunately, I don't recall if we did any benchmarking to spot check runtime/output changes, but I would expect 20 samples to be sufficient for estimating posterior means and standard deviations to the level needed for most downstream use cases.

**Comment by fleharty** [comment]
@samuelklee ahem... aren't you on vacation?

**Comment by samuelklee** [comment]
A bit of work on vacation here and there usually means I can take more vacations later!

**Comment by mwalker174** [comment]
@samuelklee Can you briefly explain how the sampling is used in CNR approximations and how they are used downstream? I would expect only 20 samples to give noisy estimates of mean and variance…

**Comment by samuelklee** [comment]
The only place I’ve used them is in the malaria genotyping method, where we calculate very simple annotations for each set of breakpoints—mostly summary statistics of the posterior means, like the minimum in a segment or the segment-level mean. We also calculate a single changepoint statistic using the posterior means. Didn’t see any indicators that posterior sampling noise was causing any issues—the only real issue I had was fixed in #7261.

I think the initial reason we started emitting these is because Talkowski folks wanted to use them for plotting/visualization, in which case I would be even less worried about sampling noise. But maybe you all are using them for something else now? Even then, not sure if anyone is using the posterior variances.

If we ever move towards emitting this for somatic/mosaicism at the single-bin, 5% level, then I might be a bit more worried—in which case, good for us for exposing it! But if you’d like to increase the default and it doesn’t seem to add significant runtime, go for it!

**Comment by mwalker174** [comment]
Thanks for that explanation. I'd rather set it to a safer value like 100 as a default - @asmirnov239 do you have any sense of how much this affects run time?

**Comment by samuelklee** [comment]
Looking at this again, plots in https://github.com/broadinstitute/gatk/issues/5764 might be useful. Probably OK to bump things up since the rest of inference will take up the bulk of runtime. Although I think one other consideration might be the WDL tests, since they run slow on Travis and already push the time limit. Perhaps set a lower value for those, if necessary? In any case, please double check all of this!

**Comment by mwalker174** [comment]
Interesting. We fixed the memory spike due to sampling correct? However, I wouldn't want to use more samples if it increases peak memory.

**Comment by samuelklee** [comment]
Yup! The sampling and calculation of means/stdevs is done online (I think @asmirnov239 added this) so it shouldn’t take additional memory to use more samples.

**Comment by asmirnov239** [comment]
I have benchmarked several runs with various parameters for number of samples for emitting denoised copy ratio. In particular, with num_samples = 20, 200, 500.
 The average shard run time for num_samples=20 was 2h34m, for num_samples=200 also 2h34m and for num_samples=500 it  was 2h35m, so there is really no significant difference in the runtime. 

Here are the concordance plots for different parameters for 3 1KG exome samples: 
![image](https://user-images.githubusercontent.com/4894545/137433767-40666bd6-bca5-4c62-913b-d518239dfab9.png)

![image](https://user-images.githubusercontent.com/4894545/137433966-b18ab68d-defc-409b-b2da-df429b48bb68.png)

![image](https://user-images.githubusercontent.com/4894545/137433981-a3c4cff3-44db-411a-8dc0-c0c9d67c39a1.png)

![image](https://user-images.githubusercontent.com/4894545/137433987-fac46d67-bf3a-4e0f-916d-2b10535f6260.png)

![image](https://user-images.githubusercontent.com/4894545/137433994-c7da1ea8-9a82-449d-8c27-7ca873f8a75e.png)

![image](https://user-images.githubusercontent.com/4894545/137433999-8767faad-41af-488b-83d3-1607c1bc27ad.png)

Since there is no effect on the runtime, and there is poor concordance for high denoised copy ratio values it makes sense to raise the default parameter to 200.
